extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -173,39 +173,45 @@ export class AnalyticsRoot {
   /**
    * Searches the element that matches the selector within the scope of the
    * analytics root in relationship to the specified context node.
-   * Note: Please wait for document-ready to avoid false missed searches with
-   * selector other than :root or :host
    *
    * @param {!Element} context
    * @param {string} selector DOM query selector.
    * @param {?string=} selectionMethod Allowed values are `null`,
    *   `'closest'` and `'scope'`.
-   * @return {?Element} Element corresponding to the selector if found.
+   * @return {Promise<?Element>} Element corresponding to the selector if found.
    */
   getElement(context, selector, selectionMethod = null) {
     // Special case selectors. The selection method is irrelavant.
+    // And no need to wait for document ready.
     if (selector == ':root') {
-      return this.getRootElement();
+      return Promise.resolve().then(() => {
+        return this.getRootElement();
+      });
     }
     if (selector == ':host') {
-      return this.getHostElement();
+      return Promise.resolve().then(() => {
```

Comment:
I used `promise.catch(error)` in testing.
However when I use `return Promise.resolve(user.assertElement(this.getHostElement()))`, I was not able to catch the error. Then I changed to `Promise.resolve().then(() => return user.assertElement(this.getHostElement()))` and then was able to catch the error. Can someone explain why this is the case?